ql/termstructures/yield/crosscurrencyswapratehelper.cpp

```diff
@@ -0,0 +1,166 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2025  Uzair Beg
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <https://www.quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/termstructures/yield/crosscurrencyswapratehelper.hpp>
+
+#include <ql/settings.hpp>
+#include <ql/time/schedule.hpp>
+#include <ql/time/dategenerationrule.hpp>
+#include <ql/currency.hpp>
+#include <ql/math/solvers1d/brent.hpp>
+#include <ql/quotes/simplequote.hpp>
+#include <algorithm>
+
+namespace QuantLib {
+
+    CrossCurrencySwapRateHelper::CrossCurrencySwapRateHelper(
+        const Handle<Quote>& fixedRate,
+        const Period& tenor,
+        const Calendar& fixedCalendar,
+        Frequency fixedFrequency,
+        BusinessDayConvention fixedConvention,
+        const DayCounter& fixedDayCount,
+        const Currency& fixedCurrency,
+        const ext::shared_ptr<IborIndex>& floatIndex,
+        const Currency& floatCurrency,
+        const Handle<Quote>& fxSpot,
```

Comment:
I don't think you need to provide the spot. It cancels out if it's a par instrument.